packages/cli/generation/ir-generator/src/converters/type-declarations/getReferencedTypesFromRawDeclaration.ts

```diff
@@ -26,7 +27,12 @@ export function getReferencedTypesFromRawDeclaration({
 }): DeclaredTypeName[] {
     const rawTypeReferences = visitRawTypeDeclaration<string[]>(typeDeclaration, {
         alias: (aliasDeclaration) => {
-            return [typeof aliasDeclaration === "string" ? aliasDeclaration : aliasDeclaration.type];
+            const aliasStr = typeof aliasDeclaration === "string" ? aliasDeclaration : aliasDeclaration.type;
+            const maybeGenericDetails = getGenericDetails(aliasStr);
+            if (maybeGenericDetails && maybeGenericDetails.isGeneric) {
+                return maybeGenericDetails.arguments ?? [];
+            }
```

Comment:
why are we returning `maybeGenericDetails.arguments ?? [];` here? 

imagine the following case: 
```yml

Node<T>: 
  properties: 
    data: T
    other: Metadata
    next: optional<string>
    prev: optional<string>

NumericNode: 
  type: Node<Datum> 
```

then `getReferencedTypesFromRawDeclaration(NumericNode)` should return `Datum` and `Metadata`

Reply:
I would expect that this part of the function builds the underlying object type declaration and calls into the object part of the visitor